### PR TITLE
Error when inserting into a table that has no `id` column

### PR DESCRIPTION
**Pull request by synbot**
### Exactly what to do

Add a property to AbstractMapper called autoIncrementColumn with default value `id`.

Change:
```php
if (! $entity->getId()) {
    $entity->setId($result->getGeneratedValue());
}
```
to:
```php
if ($this->autoIncrementColumn) {
    $entity->exchangeArray([
        $this->autoIncrementColumn => $result->getGeneratedValue()
    ]);
}
```

### CODE REVIEW

**Comment by zpchavez** [comment]
:o:

**Comment by zpchavez** [comment]
:star:

**Comment by baohx2000** [comment]
:hand:

File: src/Synapse/Mapper/InserterTrait.php [hunk 1]
```diff
@@ -58,8 +58,11 @@ protected function insertRow(AbstractEntity $entity, array $values)
 
         $result = $statement->execute();
 
-        if (! $entity->getId()) {
-            $entity->setId($result->getGeneratedValue());
+        $entityValues = $entity->getArrayCopy();
+        if ($this->autoIncrementColumn && ! $entityValues[$this->autoIncrementColumn]) {
```
Comment (paulstatezny) [inline src/Synapse/Mapper/InserterTrait.php]: I would submit that the second condition `&& ! $entityValues[$this->autoIncrementColumn]` should be omitted.

**Comment by baohx2000** [comment]
:+1:

**Comment by paulstatezny** [comment]
:o: Just opening this because I left a comment. I should have claimed this.

**Comment by zpchavez** [comment]
:star:

**Comment by spruce-bruce** [comment]
:+1: unless paul has more beef

**Comment by paulstatezny** [comment]
:sunny:

**Comment by paulstatezny** [comment]
:arrow_up:

**Pull request merged**